apix-impl/src/main/java/eu/xenit/apix/alfresco/properties/PropertyServiceImpl.java

```diff
@@ -70,6 +73,45 @@ public PropertyIndexOptions GetPropertyIndexOptions(PropertyDefinition definitio
         return propertyDefinitionUnderConstruction;
     }
 
+    public eu.xenit.apix.properties.PropertyDefinition GetPropertyDefinitionFromAlfrescoQname(
+            org.alfresco.service.namespace.QName qname) {
+
+        PropertyDefinition definition = dictionaryService.getProperty(qname);
+        eu.xenit.apix.properties.PropertyDefinition propertyDefinitionUnderConstruction = new eu.xenit.apix.properties.PropertyDefinition();
+        propertyDefinitionUnderConstruction.setName(c.apix(definition.getName()));
+        propertyDefinitionUnderConstruction.setContainer(c.apix(definition.getContainerClass().getName()));
+        propertyDefinitionUnderConstruction.setTitle(definition.getTitle());
+        propertyDefinitionUnderConstruction.setDescription(definition.getDescription());
+        propertyDefinitionUnderConstruction.setDefaultValue(definition.getDefaultValue());
+        propertyDefinitionUnderConstruction.setDataType(c.apix(definition.getDataType().getName()));
+        propertyDefinitionUnderConstruction.setMultiValued(definition.isMultiValued());
+        propertyDefinitionUnderConstruction.setMandatory(definition.isMandatory());
+        propertyDefinitionUnderConstruction.setEnforced(definition.isMandatoryEnforced());
+        propertyDefinitionUnderConstruction.setIsProtected(definition.isProtected());
+        propertyDefinitionUnderConstruction.setIndexed(this.GetPropertyIndexOptions(definition));
+        propertyDefinitionUnderConstruction.setConstraints(this.GetConstraints(definition));
+        return propertyDefinitionUnderConstruction;
+    }
```

Comment:
@rawiabenhmida This method looks very much like the method above it (GetPropertyDefinition). The only difference is the if statement and qname (alfresco <-> apix). So can you put the duplicate code in a separate method?

Reply:
Thanks for the improvement, but why not put this as a constructor for PropertyDefinition in https://github.com/xenit-eu/alfred-api/blob/master/apix-interface/src/main/java/eu/xenit/apix/properties/PropertyDefinition.java ? Or are you trying to avoid a dependency on `messageService`?

Reply:
Normally, for these conversations, the ctor takes 1 argument... but we already spent a lot of time on this, so fine for now.